src/nft/components/collection/TraitSelect.tsx

```diff
@@ -0,0 +1,191 @@
+  const toggleShowFullTraitName = useCollectionFilters((state) => state.toggleShowFullTraitName)
+
+  const { shouldShow, trait_value, trait_type } = useCollectionFilters((state) => state.showFullTraitName)
+  const isEllipsisActive = (e: MouseEvent<HTMLElement>) => {
+    if (e.currentTarget.offsetWidth < e.currentTarget.scrollWidth) {
+      toggleShowFullTraitName({
+        shouldShow: true,
+        trait_value: trait.trait_value,
+        trait_type: trait.trait_type,
+      })
+    }
+  }
+  useEffect(() => {
+    setCheckboxSelected(isTraitSelected)
+  }, [isTraitSelected])
+
+  const handleCheckbox = (e: FormEvent) => {
+    e.preventDefault()
+    scrollToTop()
+
+    if (!isCheckboxSelected) {
+      addTrait(trait)
+      setCheckboxSelected(true)
+    } else {
+      removeTrait(trait)
+      setCheckboxSelected(false)
+    }
+  }
+
+  const showFullTraitName = shouldShow && trait_type === trait.trait_type && trait_value === trait.trait_value
+
+  return (
+    <Row
+      key={trait.trait_value}
+      maxWidth="full"
+      overflowX={'hidden'}
+      overflowY={'hidden'}
+      fontWeight="normal"
+      className={`${subheadSmall} ${styles.subRowHover}`}
+      justifyContent="space-between"
+      cursor="pointer"
+      paddingLeft="12"
+      paddingRight="12"
+      style={{ paddingBottom: '21px', paddingTop: '21px', maxHeight: '44px' }}
+      onMouseEnter={handleHover}
+      onMouseLeave={handleHover}
+      onClick={handleCheckbox}
+    >
+      <Box
+        as="span"
+        whiteSpace="nowrap"
+        textOverflow="ellipsis"
+        overflow="hidden"
+        maxWidth={!showFullTraitName ? '160' : 'full'}
+        onMouseOver={(e) => isEllipsisActive(e)}
+        onMouseLeave={() => toggleShowFullTraitName({ shouldShow: false, trait_type: '', trait_value: '' })}
+      >
+        {trait.trait_type === 'Number of traits'
+          ? `${trait.trait_value} trait${pluralize(Number(trait.trait_value))}`
+          : trait.trait_value}
+      </Box>
+      <Checkbox checked={isCheckboxSelected} hovered={hovered} onChange={handleCheckbox}>
+        <Box as="span" color="darkGray" minWidth={'8'} paddingTop={'2'} paddingRight={'12'} position={'relative'}>
+          {!showFullTraitName && trait.trait_count}
+        </Box>
+      </Checkbox>
+    </Row>
+  )
+}
+
+export const TraitSelect = ({ traits, type, search }: { traits: Trait[]; type: string; search: string }) => {
+  const debouncedSearch = useDebounce(search, 300)
+
+  const addTrait = useCollectionFilters((state) => state.addTrait)
+  const removeTrait = useCollectionFilters((state) => state.removeTrait)
+  const selectedTraits = useCollectionFilters((state) => state.traits)
+
+  const [isOpen, setOpen] = useState(
+    traits.some(({ trait_type, trait_value }) => {
+      return selectedTraits.some((selectedTrait) => {
+        return selectedTrait.trait_type === trait_type && selectedTrait.trait_value === String(trait_value)
+      })
```

Comment:
nit: can remove `return` statements to `traits.some(({ trait_type, trait_value }) => selectedTraits.some((selectedTrait) => selectedTrait.trait_type === trait_type && selectedTrait.trait_value === String(trait_value))`